bin/node/executor/tests/submit_transaction.rs

```diff
@@ -98,23 +96,25 @@ fn should_submit_signed_twice_from_the_same_account() {
 	t.register_extension(TransactionPoolExt::new(pool));
 
 	let keystore = KeyStore::new();
-	keystore.write().sr25519_generate_new(Key::ID, Some(&format!("{}/hunter1", PHRASE))).unwrap();
+	keystore.write().sr25519_generate_new(AuthorityId::ID, Some(&format!("{}/hunter1", PHRASE))).unwrap();
 	t.register_extension(KeystoreExt(keystore));
 
 	t.execute_with(|| {
-		let call = pallet_balances::Call::transfer(Default::default(), Default::default());
-		let results =
-			<SubmitTransaction as SubmitSignedTransaction<Runtime, Call>>::submit_signed(call);
+		let results = Signer::<Runtime, TestAuthorityId>::all_accounts()
```

Comment:
yeah this is not correct. the first argument, `Runtime` which implements `SigningTypes` comes from the real runtime whereas the second one that defines the generic crypto types is a mock. I am surprised that it even works?

Reply:
I would document that this is the signer with the default value for the third parameter, and make a new test with `ForAny` and `ForAll` and demonstrate how the behaviour is different.